src/draw/lv_image_decoder.c

```diff
@@ -243,3 +287,59 @@ static uint32_t img_width_to_stride(lv_image_header_t * header)
     }
 }
 
+static lv_cache_compare_res_t image_decoder_cache_compare_cb(
+    const lv_image_decoder_cache_data_t * lhs,
+    const lv_image_decoder_cache_data_t * rhs)
+{
+    if(lhs->src_type == rhs->src_type) {
+        if(lhs->src_type == LV_IMAGE_SRC_FILE) {
+            int32_t cmp_res = lv_strcmp(lhs->src, rhs->src);
+            if(cmp_res != 0) {
+                return cmp_res > 0 ? 1 : -1;
+            }
+        }
+        else if(lhs->src_type == LV_IMAGE_SRC_VARIABLE) {
+            if(lhs->src != rhs->src) {
+                return lhs->src > rhs->src ? 1 : -1;
+            }
+        }
+        return 0;
+    }
+    else if(lhs->src_type == LV_IMAGE_SRC_FILE && rhs->src_type == LV_IMAGE_SRC_VARIABLE) {
+        return 1;
+    }
+    else if(lhs->src_type == LV_IMAGE_SRC_VARIABLE && rhs->src_type == LV_IMAGE_SRC_FILE) {
+        return -1;
+    }
+    return 0;
+}
+
+static void image_decoder_cache_free_cb(lv_image_decoder_cache_data_t * entry, void * user_data)
+{
+    LV_UNUSED(user_data); /*Unused*/
+
+    const lv_image_decoder_t * decoder = entry->decoder;
+    if(decoder && decoder->cache_free_cb) {
+        decoder->cache_free_cb(entry, user_data);
```

Comment:
Should store the cache_free_cb in cache instance, and use global image cache here to free.